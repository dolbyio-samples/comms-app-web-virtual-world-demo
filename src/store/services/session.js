import VoxeetSDK from "@voxeet/voxeet-web-sdk";

export const openSession = (payload) => {
  return VoxeetSDK.session
    .open(payload)
    .then((response) => ({ response }))
    .catch((error) => ({ error }));
};

export const closeSession = () => {
  return VoxeetSDK.session
    .close()
    .then((response) => ({ response }))
    .catch((error) => ({ error }));
};
