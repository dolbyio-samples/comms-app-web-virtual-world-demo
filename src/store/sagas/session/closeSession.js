import { call } from "redux-saga/effects";

import { closeSession as close } from "../../services/session";
import { displayMessage } from "../conference/message";

export default function* closeSession() {
  const { error } = yield call(close);
  if (error) {
    yield call(displayMessage, {
      type: "error",
      message: "could not close session",
    });
  }
}
