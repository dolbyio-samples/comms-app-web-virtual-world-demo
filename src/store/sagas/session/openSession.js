import { call, put } from "redux-saga/effects";
import { push } from "redux-first-history";
import { createId } from "@paralleldrive/cuid2";

import { session } from "../../effects";
import { openSession as open } from "../../services/session";
import { displayMessage } from "../conference/message";

export default function* openSession({ payload }) {
  const { error } = yield call(open, payload);
  if (error) {
    yield call(displayMessage, {
      type: "error",
      message: "could not open a session",
    });

    return;
  }

  const id = yield call(createId);
  yield put(session.opened({ id }));
  yield put(push("/conference"));
}
