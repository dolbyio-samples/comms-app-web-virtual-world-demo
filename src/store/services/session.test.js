import { describe, it, expect, vi } from "vitest";
import { openSession, closeSession } from "./session";

const mockClose = vi.fn();
const mockOpen = vi.fn();

vi.mock("@voxeet/voxeet-web-sdk", () => ({
  default: {
    get session() {
      return { close: mockClose, open: mockOpen };
    },
  },
}));

describe("session services test suite", () => {
  it("should succeed", async () => {
    // arrange
    mockOpen.mockResolvedValue({ response: {} });
    const payload = { name: "toto" };

    // act
    const { response, error } = await openSession(payload);

    // assert
    expect(mockOpen).toHaveBeenCalled();
    expect(mockOpen).toBeCalledWith(payload);
    expect(response).toMatchSnapshot();
    expect(error).toBe(undefined);
  });

  it("should succeed", async () => {
    // arrange
    mockClose.mockResolvedValue({ response: {} });

    // act
    const { result, error } = await closeSession();

    // assert
    expect(mockClose).toHaveBeenCalled();
    expect(mockClose).toHaveBeenCalled();
    expect(result).toMatchSnapshot();
    expect(error).toBe(undefined);
  });

  it("should fail and return an error", async () => {
    // arrange
    mockOpen.mockRejectedValueOnce(new Error("error"));
    const payload = { name: "toto" };

    // act
    const { response, error } = await openSession(payload);

    // assert
    expect(mockOpen).toHaveBeenCalled();
    expect(mockOpen).toBeCalledWith(payload);
    expect(error.message).toBe("error");
    expect(response).toBe(undefined);
  });

  it("should fail and return an error", async () => {
    // arrange
    mockClose.mockRejectedValueOnce(new Error("error"));

    // act
    const { response, error } = await closeSession();

    // assert
    expect(mockClose).toHaveBeenCalled();
    expect(mockClose).toHaveBeenCalled();
    expect(error.message).toBe("error");
    expect(response).toBe(undefined);
  });
});
