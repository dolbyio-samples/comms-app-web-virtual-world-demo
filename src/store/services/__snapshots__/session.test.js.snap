// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`session services test suite > should succeed 1`] = `
{
  "response": {},
}
`;

exports[`session services test suite > should succeed 2`] = `undefined`;
